### ISSUE TITLE
With default settings, PDAL can't consume its own CSV output

### ISSUE THREAD

**Issue by connormanning** [PDAL/PDAL#2201]
```
$ pdal translate https://entwine.io/data/red-rocks.laz red-rocks.csv
$ pdal info red-rocks.csv
PDAL: readers.text: Duplicate dimension ',' detected in input file 'red-rocks.csv'.
```

The reason is that the default output uses quotes around the header's dimension names.  Because the TextReader by default considers the first non-alpha/non-numeric character to be the separator, `"` is chosen, leading to the above result.

**Comment by abellgithub** [comment]
So, what should the rule be here?  What about:

"X"<sp> ,<sp>"Y"<sp> ,<sp> "Z"

or

"X""Y""Z"

or "X"<sp>"Y"<sp>"Z"

Should we just look for quoted strings, anything else be damned?

**Comment by connormanning** [comment]
> Should we just look for quoted strings, anything else be damned?

I certainly think it's worth retaining the ability to parse `X, Y, Z` without quotes which is very common.  We could do some real tokenization and handle all those cases you mentioned, although maybe at that point we are better off pulling in a library or vendoring a header implementation that handles all this for us.

**Comment by abellgithub** [comment]
I guess I wasn't clear.  I don't see changing any existing behavior.

I was going to do different parsing when the first character was a double-quote.  The question I have is what to support in that case.  It's not about a parser, it's about deciding what's acceptable.

**Comment by nicolas-chaulet** [comment]
Why do we have quotes in the header in the first place?

**Comment by abellgithub** [comment]
That's how it was originally written - one of the earliest bits of PDAL.  I'm very hesitant to change existing behavior.

CSV isn't a standard.  People use `"` to protect spaces.  Sometimes people put `"` around every field value.

**Comment by nicolas-chaulet** [comment]
Right, what do you think about the following: `"` cannot be a separator and space in between two quotes cannot be a separator either  (like `"Some field", "other"` would give `Some field` and `other` the separator being `, `)

**Comment by abellgithub** [comment]
This is still broken.

**Comment by abellgithub** [comment]
Fixed by #3129
